### PR TITLE
Alerting: Access query details of provisioned alerts

### LINKED ISSUE #57793: Provisioned alert rules are not viewable in the UI for ADX data source
Provisioned alert rules show very limited amount of information about their definition.

_**Note:** I realize that UI features may depend on the used data source type. I am using Azure Data Explorer type of data source, but the issue may be actual for other data sources as well._

image

Upon clicking _View_ button the situation does not improve a lot. In addition to the above, it shows group name (which is not very useful) and results of the queries (Stat/Table/Time Series), but not queries themselves or the used parameters (such as time range). _View in Explore_ button alleviates the situation a bit, but only for the first query. There is no way to view expressions for the subsequent queries.

image

### Reproducing steps
1. Provision an alert rule through [Provisioning API](https://grafana.com/docs/grafana/latest/developers/http_api/alerting_provisioning/) or [Terraform](https://registry.terraform.io/providers/grafana/grafana/latest/docs).
2. Navigate to the created rule in the UI.

### Expected behavior
All rule metadata is visible.

### Actual behavior
Only part of the rule definition is visible. Not enough for troubleshooting.

### Rationale
Debugging misbehaving alerts is complicated with the above limitations. Viewing/modifying a rule in the UI provides means for quick troubleshooting, whereas for provisioned alerts, the workaround seems to be creating a temporary rule in the UI, iteratively modifying it until its JSON definition matches to the definition of the provisioned (buggy) rule, and then figuring out what's wrong. This is a tedious process.

### Proposed solutions (alternatives)
1. **Make provisioned rules editable in the UI.**
Seriously, what is the point? If you are using Terraform, it will synchronize the rule with the IaC definition on the next run or is able to just show differences. If using Provisioning API, each call provides definition of the entire alert rule, I do not see the room for conflicts. (And if the user **wants** to shoot into their leg, why don't allow them?)
2. **Make viewing UI (upon clicking _View_ button) same as Edit UI, just read-only.**
All defining parameters must be visible. This way it at least simplifies creating of the debugging rule.
3. **Make an ability to duplicate a rule in the UI (in non-provisioned state).**

### Environment
- Grafana version: Cloud
- Data source type & version: Azure Data Explorer, v4.1.9
- OS Grafana is installed on: N/A
- User OS & Browser: Chrome
- Grafana plugins: Azure Data Explorer, v4.1.9

### PR DESCRIPTION
**Pull request by konrad147**
**What is this feature?**
This PR adds a query viewer on the alert rule view page

Fixes #57793

**Special notes for your reviewer**:
Query editors are provided by data source plugins and usually, they don't provide a `read-only` mode. 
Due to that fact, we decided to provide code-based viewer for queries and create read-only versions of expressions

**Grafana Managed Rules**
image

**Cloud Rules**
image

**Provisioned Rules**
image

### CODE REVIEW

File: public/app/features/alerting/unified/RuleViewer.tsx [hunk 1]
```diff
@@ -183,33 +197,49 @@ export function RuleViewer({ match }: RuleViewerProps) {
...
+            queries={queries}
+            evalDataByQuery={data}
+            evalTimeRanges={evaluationTimeRanges}
+            onTimeRangeChange={onQueryTimeRangeChange}
+          />
+        )}
+
+        {!isGrafanaRulerRule(rule.rulerRule) && !isFederatedRule && data && Object.keys(data).length > 0 && (
+          <div className={styles.queries}>
+            {queries.map((query) => {
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/RuleViewer.tsx:220]: I know that probably having an array this is the only way, but can we have more that 1 query in here?

File: public/app/features/alerting/unified/components/expressions/Expression.tsx [hunk 2]
```diff
@@ -109,24 +105,7 @@ export const Expression: FC<ExpressionProps> = ({
...
-            {!emptyResults &&
-              !isTimeSeriesResults &&
-              series.map((frame, index) => (
-                // There's no way to uniquely identify a frame that doesn't cause render bugs :/ (Gilles)
-                <FrameRow key={uniqueId()} frame={frame} index={index} isAlertCondition={alertCondition} />
-              ))}
-            {emptyResults && <div className={cx(styles.expression.noData, styles.mutedText)}>No data</div>}
-          </div>
-        )}
+        {hasResults && <ExpressionResult series={series} isAlertCondition={isAlertCondition} />}
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/expressions/Expression.tsx:108]: ❤️

**Review by soniaAguilarPeiron — APPROVED** [review]
Great job! 🎉
I added some non-blocking comments.

**Comment by gillesdemey** [comment]
I think we need some additional CSS styling to accommodate long label keys, I've attached some screenshots with "before" and "after" some CSS changes.

I'll append these to the PR since I've already sort of figured out where they need to go and I have some test data :)

**Before**
Screenshot 2022-12-12 at 13 33 15

**After**
Screenshot 2022-12-12 at 13 33 01

File: public/app/features/alerting/unified/RuleViewer.tsx (hunk 1 above)
Comment (konrad147) [reply public/app/features/alerting/unified/RuleViewer.tsx:220]: No, there should be only one query for cloud rules

**Review by gillesdemey — APPROVED** [review]
Awesome work 🚀

**Pull request merged**